pensieve/analysis.py

```diff
@@ -85,23 +85,34 @@ def _get_timelimits_if_ready(
                 return None
 
             return current_time_limits
+        elif period == AnalysisPeriod.OVERALL:
+            # The last full day of data for an experiment is the day before an operator
+            # switches it off.
+            tomorrow = current_date + timedelta(days=1)
+            if self.config.experiment.end_date != tomorrow:
+                return None
 
-        # Period is OVERALL
-        # The last full day of data for an experiment is the day before an operator switches it off.
-        tomorrow = current_date + timedelta(days=1)
-        if self.config.experiment.end_date != tomorrow:
-            return None
-
-        return TimeLimits.for_single_analysis_window(
-            last_date_full_data=current_date_str,
-            analysis_start_days=0,
-            analysis_length_dates=(
-                self.config.experiment.end_date - self.config.experiment.start_date
-            ).days
-            - dates_enrollment
-            + 1,
-            **time_limits_args,
-        )
+            return TimeLimits.for_single_analysis_window(
+                last_date_full_data=current_date_str,
+                analysis_start_days=0,
+                analysis_length_dates=(
+                    self.config.experiment.end_date - self.config.experiment.start_date
+                ).days
+                - dates_enrollment
+                + 1,
+                **time_limits_args,
+            )
+        elif period == AnalysisPeriod.ENROLLMENT:
+            if ((current_date - self.config.experiment.start_date).days + 1) != dates_enrollment:
+                return None
+            return TimeLimits.for_single_analysis_window(
+                last_date_full_data=current_date_str,
+                analysis_start_days=0,
+                analysis_length_dates=1,
```

Comment:
Just double checking, is the analysis window supposed to be 1 day long or should it cover the entire enrollment period?

Reply:
This is an interesting question that I was trying to avoid; thanks for calling it out!

I wrote out a doc and convinced myself that this is actually the wrong approach; this needs to encompass the entire enrollment period, not a fixed number of days from a client's enrollment.

That doesn't match the mozanalysis metric paradigm, so these need to be computed in the custom enrollments query that CFRs need anyway.

Then we need to figure out how to summarize them 🤔

Reply:
Hm, no, now I'm convinced we need to wait to measure CTR until we stop showing impressions, which I think isn't until the experiment actually ends. Should figure out what users will accept here

Reply:
Resolved:
- Final numbers are available once the experiment ends since intervention delivery isn't complete until the experiment ends
- Preliminary numbers can be tracked in monitoring if people insist
- There's no need for an enrollments window